### PR TITLE
Feature/tao 2665/qti runner feedback forms

### PR DESCRIPTION
**Pull request by zagovorichev**
https://oat-sa.atlassian.net/browse/TAO-2665

### CODE REVIEW

File: views/js/runner/provider/qti.js [hunk 1]
```diff
@@ -253,6 +207,23 @@ define([
...
+            var self = this;
+
+            var _renderer = self._item.getRenderer();
+            var _loader   = new QtiLoader(self._item);
+
+            // loading feedbacks from response into the current item
+            _loader.loadElements(feedbacks, function (item) {
+                _renderer.load(function () {
+
+                    var renderingQueue = modalFeedbackHelper.getFeedbacks(item, itemSession);
```
Comment (krampstudio) [inline views/js/runner/provider/qti.js]: Why do you use an helper when you can do `feedback.render()` ?

**Comment by zagovorichev** [comment]
depends on https://github.com/oat-sa/extension-tao-item/pull/231

File: views/js/runner/provider/qti.js (hunk 1 above)
Comment (zagovorichev) [inline views/js/runner/provider/qti.js]: feedback.render() - create html from object 
modalFeedbackHelper.getFeedbacks() - creates objects with information how and where to insert feedback

File: views/js/qtiItem/helper/modalFeedback.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+     * @returns {Array}
+     */
+    var getInteractionsDisplayInfo = function getInteractionsDisplayInfo(item) {
+
+        var interactionsDisplayInfo = [];
+        var $itemContainer = item.getContainer();
+        var interactionOrder = 0;
+
+        //extract all interaction related information needed to display their
+        _.each(item.getComposingElements(), function (element) {
```
Comment (krampstudio) [inline views/js/qtiItem/helper/modalFeedback.js]: Please use `_.forEach` (we want to limit the use of aliases to facilitate the migration to the next version of lodash)

File: views/js/qtiItem/helper/modalFeedback.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            var feedbackIds, message, $container, comparedOutcome, _currentMessageGroupId, interactionInfo;
+            var outcomeIdentifier = feedback.attr('outcomeIdentifier');
+            var order = -1;
+
+            //verify if the feedback should be displayed
+            if (itemSession[outcomeIdentifier]) {
+
+                //is the feedback in the list of feedbacks to be displayed ?
+                feedbackIds = pci.getRawValues(itemSession[outcomeIdentifier]);
+                if (_.indexOf(feedbackIds, feedback.id()) === -1) {
```
Comment (krampstudio) [inline views/js/qtiItem/helper/modalFeedback.js]: easiest to read with `_.contains`

File: views/js/qtiItem/helper/modalFeedback.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                }
+            });
+            interactionOrder++;
+        });
+        interactionsDisplayInfo = _.sortBy(interactionsDisplayInfo, 'order');
+
+        return interactionsDisplayInfo;
+    };
+
+    var getFeedbacks = function getFeedbacks(item, itemSession) {
```
Comment (krampstudio) [inline views/js/qtiItem/helper/modalFeedback.js]: Missing doc

**Pull request merged**